### PR TITLE
Restored pushing thermostat data over the heartbeat

### PR DESCRIPTION
**Pull request by khenderick**

### CODE REVIEW

File: src/vpn_service.py [hunk 1]
```diff
@@ -189,6 +189,12 @@ def get_pulse_counter_diff(self):
             return ret
         return
 
+    @staticmethod
+    def __counter_diff(current, previous):
+        """ Calculate the diff between two counter values. """
+        diff = current - previous
+        return diff if diff >= 0 else 65536 - previous + current
```
Comment (svanscho) [inline src/vpn_service.py]: else 65536 + diff ?

File: src/vpn_service.py [hunk 2]
```diff
@@ -200,11 +206,22 @@ def get_enabled_outputs(self):
...
+        if data is None or data['success'] is False:
+            return None
+        ret = {'thermostats_on': data['thermostats_on'],
+               'automatic': data['automatic'],
+               'cooling': data['cooling']}
+        thermostats = []
+        for thermostat in data['status']:
+            to_add = {}
+            for field in ['id', 'act', 'csetp', 'mode', 'output0', 'output1', 'outside', 'airco']:
+                to_add[field] = thermostat[field]
```
Comment (svanscho) [inline src/vpn_service.py]: to_add[field] = thermosta.get(field, None) ?

Comment (svanscho) [reply src/vpn_service.py]: As agreed, let's keep changes to a minimum to be as backwards compatible as possible.

File: src/vpn_service.py (hunk 1 above)
Comment (svanscho) [reply src/vpn_service.py]: Ok for now.

Comment (khenderick) [reply src/vpn_service.py]: Good point, but I don't want to change unaffected code (it just has been moved causing it to show up as a difference). All new flows use metrics, so it's obsolete code pending deletion anyway.

File: src/vpn_service.py (hunk 2 above)
Comment (khenderick) [reply src/vpn_service.py]: We're quite sure these fields will be in there, and I'd rather have a key error here, than sending unexpected data to the cloud in this case.

**Pull request merged**